server/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -104,6 +106,22 @@ private RestRequest(NamedXContentRegistry xContentRegistry, Map<String, String>
         this.requestId = requestId;
     }
 
+    private static @Nullable ParsedMediaType parseHeaderWithMediaType(Map<String, List<String>> headers, String headerName) {
+        //TOOD: shouldn't this be case insensitive ?
```

Comment:
should we resolve this TODO with this PR ?

Reply:
I think we should. this would apply to all usages of headers. RFC says that it should be case-insensitive